include/Memory.h

```diff
@@ -31,25 +32,36 @@
 #include <vector>
 
 #include "lmms_export.h"
+#include "NiftyCounter.h"
 
 class LMMS_EXPORT MemoryManager
 {
 public:
-	struct ThreadGuard
-	{
-		ThreadGuard();
-		~ThreadGuard();
-	};
-
 	static void * alloc( size_t size );
 	static void free( void * ptr );
+
+private:
+	static void initialize();
+	static void deinitialize();
+	static void thread_initialize();
+	static void thread_deinitialize();
+public:
+	typedef NiftyCounter<MemoryManager::initialize, MemoryManager::deinitialize> MmCounter;
+	typedef NiftyCounterTL<MemoryManager::thread_initialize, MemoryManager::thread_deinitialize> ThreadGuard;
 };
 
+static MemoryManager::MmCounter _mm_counter;
+static thread_local MemoryManager::MmCounter _mm_thread_counter;
```

Comment:
Why do you have thread local `MmCounter` here and `ThreadGuard` in `Memory.cpp`?

Reply:
Sorry for missing this question. I don't remember exactly to be honest. The `MmCounter` was introduced in ac0081de10dc1c494c8366b93cc3e0cc9e275a03 to fix a crash in MSVC, IIRC it would call `rpmalloc_finalize` before `rpmalloc_thread_finalize`. I realize this whole Nifty Counter solution may be a bit fragile to use, so I'm open to searching a different solution if we can't make it work reliably.